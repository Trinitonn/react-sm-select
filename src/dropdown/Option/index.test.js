import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Option } from './index';

jest.mock('../../utils');

const methods = {
  onClick: jest.fn(),
};

describe('Option', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Option
        option={{ value: 'one', label: 'One' }}
        checked={true}
        focused={true}
        isSingle={false}
        {...methods}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render not focused', () => {
    const wrapper = shallow(
      <Option
        option={{ value: 'one', label: 'One' }}
        checked={false}
        focused={false}
        isSingle={true}
        {...methods}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly with custom Option', () => {
    const wrapper = shallow(
      <Option
        Option={() => (<div>Custom Option</div>)}
        option={{ value: 'one', label: 'One' }}
        checked={true}
        focused={true}
        isSingle={false}
        {...methods}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('updateFocus should focus if focused', () => {
    const wrapper = shallow(
      <Option
        option={{ value: 'one', label: 'One' }}
        checked={true}
        focused={true}
        isSingle={false}
        {...methods}
      />
    );
    wrapper.instance().optionRef = {
      current: {
        focus: jest.fn(),
      }
    };
    wrapper.instance().updateFocus();

    expect(wrapper.instance().optionRef.current.focus).toHaveBeenCalled();
  });

  it('updateFocus should not focus if not focused', () => {
    const wrapper = shallow(
      <Option
        option={{ value: 'one', label: 'One' }}
        checked={true}
        focused={false}
        isSingle={false}
        {...methods}
      />
    );
    wrapper.instance().optionRef = {
      current: {
        focus: jest.fn(),
      }
    };
    wrapper.instance().updateFocus();

    expect(wrapper.instance().optionRef.current.focus).not.toHaveBeenCalled();
  });

  it('handleKeyDown should handle Enter press', () => {
    const wrapper = shallow(
      <Option
        option={{ value: 'one', label: 'One' }}
        checked={true}
        focused={false}
        isSingle={false}
        {...methods}
      />
    );
    wrapper.instance().handleKeyDown({ which: 13 });

    expect(methods.onClick).toHaveBeenCalled();
  });

  it('componentDidMount should call updateFocus', () => {
    const wrapper = shallow(
      <Option
        option={{ value: 'one', label: 'One' }}
        checked={true}
        focused={false}
        isSingle={false}
        {...methods}
      />
    );
    const updateFocus = jest.spyOn(wrapper.instance(), 'updateFocus');
    wrapper.instance().componentDidMount();

    expect(updateFocus).toHaveBeenCalled();
  });

  it('componentDidUpdate should call updateFocus', () => {
    const wrapper = shallow(
      <Option
        option={{ value: 'one', label: 'One' }}
        checked={true}
        focused={false}
        isSingle={false}
        {...methods}
      />
    );
    const updateFocus = jest.spyOn(wrapper.instance(), 'updateFocus');
    wrapper.instance().componentDidUpdate();

    expect(updateFocus).toHaveBeenCalled();
  });
});
