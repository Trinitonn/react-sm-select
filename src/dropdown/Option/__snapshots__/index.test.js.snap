// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should render not focused 1`] = `
<label
  aria-selected={false}
  onKeyDown={[Function]}
  onMouseDown={[MockFunction]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  role="option"
  tabIndex="-1"
>
  <DefOption
    checked={false}
    isSingle={true}
    option={
      Object {
        "label": "One",
        "value": "one",
      }
    }
  />
</label>
`;

exports[`Option should render properly 1`] = `
<label
  aria-selected={true}
  onKeyDown={[Function]}
  onMouseDown={[MockFunction]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  role="option"
  tabIndex="-1"
>
  <DefOption
    checked={true}
    isSingle={false}
    option={
      Object {
        "label": "One",
        "value": "one",
      }
    }
  />
</label>
`;

exports[`Option should render properly with custom Option 1`] = `
<label
  aria-selected={true}
  onKeyDown={[Function]}
  onMouseDown={[MockFunction]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  role="option"
  tabIndex="-1"
>
  <Option
    checked={true}
    isSingle={false}
    option={
      Object {
        "label": "One",
        "value": "one",
      }
    }
  />
</label>
`;
